clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRecords.java

```diff
@@ -53,6 +60,14 @@ public List<ConsumerRecord<K, V>> records(TopicPartition partition) {
             return Collections.unmodifiableList(recs);
     }
 
+    /**
+     * Get the next offsets and metadata corresponding to all partitions fetched from the last batch.
```

Comment:
Why `fetched from the last batch`? When I follow the code, it seems the result can include the offsets from multiple batches?

@kirktrue made a fair point that other comments are also not good in this class. But for this method specifically, I think it should be clear for which topic-partitions we include offsets and for which topic-partitions we don't, as this will define what offsets we commit, and committing tends to be important in Kafka. So it should be very clear what is the set of offsets returned by this method.

I think we can say that we include offsets for all topic-partitions for which the position advanced in this poll call. But then, we need to change the code a bit (below). 

For me, this is not a minor point.